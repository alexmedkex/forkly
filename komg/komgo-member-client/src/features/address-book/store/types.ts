import { List } from 'immutable'
import { ImmutableMap } from '../../../utils/types'
import { Action } from 'redux'
import { IMember } from '../../members/store/types'

export enum AddressBookActionType {
  GET_COMPANIES_REQUEST = '@@addressBook/GET_COMPANIESS_REQUEST',
  GET_COMPANIES_SUCCESS = '@@addressBook/GET_COMPANIESS_SUCCESS',
  GET_COMPANIES_FAILURE = '@@addressBook/GET_COMPANIESS_FAILURE',
  GET_COMPANY_REQUEST = '@@addressBook/GET_COMPANY_REQUEST',
  GET_COMPANY_SUCCESS = '@@addressBook/GET_COMPANY_SUCCESS',
  GET_COMPANY_FAILURE = '@@addressBook/GET_COMPANY_FAILURE',
  GENERATE_MEMBER_PACKAGE_REQUEST = '@@addressBook/GENERATE_MEMBER_PACKAGE_REQUEST',
  GENERATE_MEMBER_PACKAGE_SUCCESS = '@@addressBook/GENERATE_MEMBER_PACKAGE_SUCCESS',
  GENERATE_MEMBER_PACKAGE_FAILURE = '@@addressBook/GENERATE_MEMBER_PACKAGE_FAILURE',
  ADD_COMPANY_TO_ENS_REQUEST = '@@addressBook/ADD_COMPANY_TO_ENS_REQUEST',
  ADD_COMPANY_TO_ENS_SUCCESS = '@@addressBook/ADD_COMPANY_TO_ENS_SUCCESS',
  ADD_COMPANY_TO_ENS_FAILURE = '@@addressBook/ADD_COMPANY_TO_ENS_FAILURE',
  CONFIGURE_MQ_REQUEST = '@@addressBook/CONFIGURE_MQ_REQUEST',
  CONFIGURE_MQ_SUCCESS = '@@addressBook/CONFIGURE_MQ_SUCCESS',
  CONFIGURE_MQ_FAILURE = '@@addressBook/CONFIGURE_MQ_FAILURE',
  CREATE_COMPANY_REQUEST = '@@addressBook/CREATE_COMPANY_REQUEST',
  CREATE_COMPANY_SUCCESS = '@@addressBook/CREATE_COMPANY_SUCCESS',
  CREATE_COMPANY_FAILURE = '@@addressBook/CREATE_COMPANY_FAILURE',
  UPDATE_COMPANY_REQUEST = '@@addressBook/UPDATE_COMPANY_REQUEST',
  UPDATE_COMPANY_SUCCESS = '@@addressBook/UPDATE_COMPANY_SUCCESS',
  UPDATE_COMPANY_FAILURE = '@@addressBook/UPDATE_COMPANY_FAILURE',
  DeleteCompanyByIdSuccess = '@@addressBook/DELETE_COMPANY_BY_ID_SUCCESS'
}

export interface AddressBookStateProperties {
  companies: List<IMember>
}

export type AddressBookState = ImmutableMap<AddressBookStateProperties>

export interface GetCompaniesRequest extends Action {
  type: AddressBookActionType.GET_COMPANIES_SUCCESS
  payload: IMember[]
}
export interface GetCompanyRequest extends Action {
  type: AddressBookActionType.GET_COMPANY_SUCCESS
  payload: IMember
}
export interface GenerateMemberPackageRequest extends Action {
  type: AddressBookActionType.GENERATE_MEMBER_PACKAGE_SUCCESS
  payload: IMember
}
export interface AddCompanyToENSRequest extends Action {
  type: AddressBookActionType.ADD_COMPANY_TO_ENS_SUCCESS
  payload: IMember
}
export interface ConfigureMQRequest extends Action {
  type: AddressBookActionType.CONFIGURE_MQ_SUCCESS
  payload: IMember
}

export interface CreateCompanyRequest extends Action {
  type: AddressBookActionType.CREATE_COMPANY_SUCCESS
  payload: IMember
}
export interface UpdateCompanyRequest extends Action {
  type: AddressBookActionType.UPDATE_COMPANY_SUCCESS
  payload: IMember
}

export interface DeleteCompanyByIdSuccess extends Action {
  type: AddressBookActionType.DeleteCompanyByIdSuccess
  payload: { id: string }
}

export type AddressBookActions =
  | GetCompaniesRequest
  | GetCompanyRequest
  | CreateCompanyRequest
  | UpdateCompanyRequest
  | GenerateMemberPackageRequest
  | AddCompanyToENSRequest
  | ConfigureMQRequest
  | DeleteCompanyByIdSuccess
