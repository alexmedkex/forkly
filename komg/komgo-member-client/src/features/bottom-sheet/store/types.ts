import { Action } from 'redux'
import { ImmutableMap } from '../../../utils/types'
import {
  CreateDocumentError,
  CreateDocumentSuccess,
  DocumentRegisteredError,
  DocumentRegisteredSuccess,
  HasId,
  HasName
} from '../../document-management/store/types'
import { ActionCreator } from 'react-redux'

export enum BottomSheetActionType {
  REMOVE_BOTTOMSHEET_ITEM = '@@btsh/REMOVE_BOTTOMSHEET_ITEM',
  RETRY_BOTTOMSHEET_ITEM = '@@btsh/RETRY_BOTTOMSHEET_ITEM',
  UPDATE_BOTTOMSHEET_ITEM = '@@btsh/UPDATE_BOTTOMSHEET_ITEM',
  CREATE_BOTTOMSHEET_ITEM = '@@btsh/CREATE_BOTTOMSHEET_ITEM'
}

export interface BottomSheetItem extends HasId, HasName {
  state: BottomSheetStatus
  itemType?: BottomSheetItemType
  displayStatus?: string
  action?: ActionCreator<any>
  navigationLink?: string
}

export enum BottomSheetItemType {
  REGISTER_KYC_DOCUMENT,
  CREATE_MAGIC_LINK,
  REGISTER_RD_DOCUMENT,
  REGISTER_TRADE_DOCUMENT
}

export enum BottomSheetStatus {
  NEW = 'NEW',
  PENDING = 'PENDING',
  REGISTERED = 'REGISTERED',
  FAILED = 'FAILED'
}

export interface BottomSheetStateFields {
  visible: boolean
  items: BottomSheetItem[]
}

export interface RemoveBottomSheetItem extends Action {
  type: BottomSheetActionType.REMOVE_BOTTOMSHEET_ITEM
  payload: HasId
}

export interface RetryRegisterItem extends Action {
  type: BottomSheetActionType.RETRY_BOTTOMSHEET_ITEM
  payload: BottomSheetItem
}

export interface UpdateBottomSheetItem extends Action {
  type: BottomSheetActionType.UPDATE_BOTTOMSHEET_ITEM
  payload: BottomSheetItem
}

export interface CreateBottomSheetItem extends Action {
  type: BottomSheetActionType.CREATE_BOTTOMSHEET_ITEM
  payload: BottomSheetItem
}

export type BottomSheetState = ImmutableMap<BottomSheetStateFields>

export type BottomSheetAction =
  | DocumentRegisteredSuccess
  | DocumentRegisteredError
  | CreateDocumentSuccess
  | CreateDocumentError
  | RemoveBottomSheetItem
  | RetryRegisterItem
  | UpdateBottomSheetItem
  | CreateBottomSheetItem
