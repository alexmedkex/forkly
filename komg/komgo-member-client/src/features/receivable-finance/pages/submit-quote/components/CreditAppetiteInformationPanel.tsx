import React from 'react'
import { CounterpartyCreditLineDataView } from '../../../../credit-line/components/financial-institution/credit-line-view/CounterpartyCreditLineDataView'
import { CreditLineType, IExtendedCreditLine, IExtendedSharedCreditLine } from '../../../../credit-line/store/types'
import { BoldHeaderWrapper } from '../../../../../components/styled-components'
import styled from 'styled-components'
import { grey, SPACES } from '@komgo/ui-components'
import { SharedCreditLineDataView } from '../../../../credit-line/components/financial-institution/credit-line-view/SharedCreditLineDataView'

export interface ICreditAppetiteInformationPanelProps {
  creditLine: IExtendedCreditLine
  buyerName: string
  sellerName: string
  sellerStaticId: string
}

export const CreditAppetiteInformationPanel: React.FC<ICreditAppetiteInformationPanelProps> = props => {
  const { creditLine, buyerName, sellerName, sellerStaticId } = props

  if (!creditLine) {
    return (
      <>
        <BoldHeaderWrapper data-test-id="credit-appetite-header-empty">
          <h3>No credit appetite information on {buyerName}</h3>
          <p>To enter information, go to the appetite section and add {buyerName}.</p>
        </BoldHeaderWrapper>
      </>
    )
  }

  const buyerTitle = `${buyerName} information`
  const buyerText = 'Internal information relating to the buyer'

  const sellerTitle = `${sellerName} information`
  const sellerText = 'Information you have shared with specific sellers'

  const sellerSharedCreditLines = creditLine.sharedCreditLines.filter(
    (v: IExtendedSharedCreditLine) => v.sharedWithStaticId === sellerStaticId
  )
  const hasSellerSharedCreditLine = sellerSharedCreditLines.length > 0

  return (
    <>
      <BoldHeaderWrapper data-test-id="credit-appetite-header-buyer">
        <h3>{buyerTitle}</h3>
        <p>{buyerText}</p>
      </BoldHeaderWrapper>
      <CounterpartyCreditLineDataView
        creditLine={creditLine}
        feature={CreditLineType.RiskCover}
        showFieldVerticalDisplay={true}
        hideOptionalLabel={true}
        data-test-id="credit-appetite-data-buyer"
      />
      {hasSellerSharedCreditLine && (
        <SellerSection>
          <BoldHeaderWrapper data-test-id="credit-appetite-header-seller">
            <h3>{sellerTitle}</h3>
            <p>{sellerText}</p>
          </BoldHeaderWrapper>
          <SharedDataSection>
            <SharedCreditLineDataView
              sharedCreditLine={sellerSharedCreditLines[0]}
              feature={CreditLineType.RiskCover}
              data-test-id="credit-appetite-data-seller"
            />
          </SharedDataSection>
        </SellerSection>
      )}
    </>
  )
}

const SellerSection = styled.div`
  border-top: 1px solid ${grey};
  padding: 20px 0;

  && {
    .column:not(.grid) {
      padding-bottom: 10px;
    }
  }
`

const SharedDataSection = styled.div`
  margin-top: ${SPACES.SMALL};
`
