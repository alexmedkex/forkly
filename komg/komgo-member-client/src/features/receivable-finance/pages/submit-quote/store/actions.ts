import { ActionCreator } from 'react-redux'
import { Action } from 'redux'

import {
  ReceivableDiscountingActionThunk,
  fetchDiscountingRequest
} from '../../../../receivable-discounting-legacy/store/application/actions'
import { fetchCreditLines } from '../../../../credit-line/store/actions'
import { Products } from '../../../../document-management/constants/Products'
import { SubProducts } from '../../../../document-management/constants/SubProducts'

export const fetchRDRequesForSubmitQuote: ActionCreator<ReceivableDiscountingActionThunk> = (rdId: string) => (
  dispatch,
  _,
  api
): Action => {
  fetchCreditLines(Products.TradeFinance, SubProducts.ReceivableDiscounting)(dispatch, _ as any, api)
  return fetchDiscountingRequest(rdId)(dispatch, _, api)
}
