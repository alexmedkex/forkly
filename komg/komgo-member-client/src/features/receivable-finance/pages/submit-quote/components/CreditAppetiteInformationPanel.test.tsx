import React from 'react'
import { render } from '@testing-library/react'
import { buildFakeRiskCoverData, buildFakeSharedCreditLine, buildFakeCreditLine } from '@komgo/types'
import moment from 'moment-timezone'
import { ICreditAppetiteInformationPanelProps, CreditAppetiteInformationPanel } from './CreditAppetiteInformationPanel'
import { IExtendedCreditLine } from '../../../../credit-line/store/types'

describe('CreditAppetiteInformationPanel', () => {
  let testProps: ICreditAppetiteInformationPanelProps

  beforeEach(() => {
    moment.tz.guess = jest.fn(() => 'Europe/Belgrade')
    moment.tz.setDefault('Europe/Belgrade')

    const extendedCreditLine: IExtendedCreditLine = {
      ...buildFakeCreditLine(),
      counterpartyName: 'BP',
      counterpartyLocation: '',
      data: buildFakeRiskCoverData(),
      sharedCreditLines: [{ ...buildFakeSharedCreditLine(), counterpartyName: '' }]
    }

    testProps = {
      creditLine: extendedCreditLine,
      sellerStaticId: 'sellerStaticId',
      buyerName: 'Mr. Buyer',
      sellerName: 'Ms. Seller'
    }
  })

  it('should render correctly only Buyer information', () => {
    expect(render(<CreditAppetiteInformationPanel {...testProps} />).asFragment()).toMatchSnapshot()
  })

  it('should render correctly no credit information', () => {
    testProps.creditLine = undefined

    expect(render(<CreditAppetiteInformationPanel {...testProps} />).asFragment()).toMatchSnapshot()
  })

  it('should render correctly Buyer and Seller information', () => {
    testProps.creditLine.sharedCreditLines[0].sharedWithStaticId = testProps.sellerStaticId

    expect(render(<CreditAppetiteInformationPanel {...testProps} />).asFragment()).toMatchSnapshot()
  })
})
