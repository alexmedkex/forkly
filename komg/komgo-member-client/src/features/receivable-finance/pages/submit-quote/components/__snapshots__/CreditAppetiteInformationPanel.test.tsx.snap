// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreditAppetiteInformationPanel should render correctly Buyer and Seller information 1`] = `
<DocumentFragment>
  .c1 h3 {
  margin-bottom: 0px;
}

.c1 p {
  font-style: italic;
  width: 80%;
}

.c0.c0.c0.c0 h3 {
  font-weight: bold;
}

<div
    class="c0 c1"
    data-test-id="credit-appetite-header-buyer"
  >
    <h3>
      Mr. Buyer information
    </h3>
    <p>
      Internal information relating to the buyer
    </p>
  </div>
  .c2 {
  display: inline-block;
  font-weight: bold;
}

.c0 {
  margin-top: 24px;
  margin-bottom: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0:last-child {
  margin-bottom: 0px;
}

.c1 {
  width: 250px;
  margin-right: 10px;
}

.c3 {
  display: inline-block;
}

@media (max-width:768px) {
  .c0 {
    -webkit-flex-flow: column;
    -ms-flex-flow: column;
    flex-flow: column;
    -webkit-align-items: unset;
    -webkit-box-align: unset;
    -ms-flex-align: unset;
    align-items: unset;
  }
}

<div
    class="c0"
    label="Appetite"
    value="Yes"
  >
    <div
      class="c1"
      data-test-id="appetite-label"
    >
      <span
        class="c2"
      >
        Appetite 
      </span>
    </div>
    <div
      class="c3"
      data-test-id="appetite-value"
    >
      Yes
    </div>
  </div>
  .c2 {
  display: inline-block;
  font-weight: bold;
}

.c0 {
  margin-top: 24px;
  margin-bottom: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0:last-child {
  margin-bottom: 0px;
}

.c1 {
  width: 250px;
  margin-right: 10px;
}

.c3 {
  display: inline-block;
}

@media (max-width:768px) {
  .c0 {
    -webkit-flex-flow: column;
    -ms-flex-flow: column;
    flex-flow: column;
    -webkit-align-items: unset;
    -webkit-box-align: unset;
    -ms-flex-align: unset;
    align-items: unset;
  }
}

<div
    class="c0"
    label="Credit limit"
    value="$ 10,000,000"
  >
    <div
      class="c1"
      data-test-id="credit-limit-label"
    >
      <span
        class="c2"
      >
        Credit limit 
      </span>
    </div>
    <div
      class="c3"
      data-test-id="credit-limit-value"
    >
      $ 10,000,000
    </div>
  </div>
  .c2 {
  display: inline-block;
  font-weight: bold;
}

.c0 {
  margin-top: 24px;
  margin-bottom: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0:last-child {
  margin-bottom: 0px;
}

.c1 {
  width: 250px;
  margin-right: 10px;
}

.c3 {
  display: inline-block;
}

@media (max-width:768px) {
  .c0 {
    -webkit-flex-flow: column;
    -ms-flex-flow: column;
    flex-flow: column;
    -webkit-align-items: unset;
    -webkit-box-align: unset;
    -ms-flex-align: unset;
    align-items: unset;
  }
}

<div
    class="c0"
    label="Expiry date"
  >
    <div
      class="c1"
      data-test-id="expiry-date-label"
    >
      <span
        class="c2"
      >
        Expiry date 
      </span>
    </div>
    <div
      class="c3"
      data-test-id="expiry-date-value"
    >
      -
    </div>
  </div>
  .c2 {
  display: inline-block;
  font-weight: bold;
}

.c0 {
  margin-top: 24px;
  margin-bottom: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0:last-child {
  margin-bottom: 0px;
}

.c1 {
  width: 250px;
  margin-right: 10px;
}

.c3 {
  display: inline-block;
}

@media (max-width:768px) {
  .c0 {
    -webkit-flex-flow: column;
    -ms-flex-flow: column;
    flex-flow: column;
    -webkit-align-items: unset;
    -webkit-box-align: unset;
    -ms-flex-align: unset;
    align-items: unset;
  }
}

<div
    class="c0"
    label="Availability"
    value="Yes"
  >
    <div
      class="c1"
      data-test-id="availability-label"
    >
      <span
        class="c2"
      >
        Availability 
      </span>
    </div>
    <div
      class="c3"
      data-test-id="availability-value"
    >
      Yes
    </div>
  </div>
  .c2 {
  display: inline-block;
  font-weight: bold;
}

.c0 {
  margin-top: 24px;
  margin-bottom: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0:last-child {
  margin-bottom: 0px;
}

.c1 {
  width: 250px;
  margin-right: 10px;
}

.c3 {
  display: inline-block;
}

@media (max-width:768px) {
  .c0 {
    -webkit-flex-flow: column;
    -ms-flex-flow: column;
    flex-flow: column;
    -webkit-align-items: unset;
    -webkit-box-align: unset;
    -ms-flex-align: unset;
    align-items: unset;
  }
}

<div
    class="c0"
    label="Amount available"
  >
    <div
      class="c1"
      data-test-id="amount-available-label"
    >
      <span
        class="c2"
      >
        Amount available 
      </span>
    </div>
    <div
      class="c3"
      data-test-id="amount-available-value"
    >
      <span>
        $ 8,000,000
      </span>
    </div>
  </div>
  .c2 {
  display: inline-block;
  font-weight: bold;
}

.c0 {
  margin-top: 24px;
  margin-bottom: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0:last-child {
  margin-bottom: 0px;
}

.c1 {
  width: 250px;
  margin-right: 10px;
}

.c3 {
  display: inline-block;
}

@media (max-width:768px) {
  .c0 {
    -webkit-flex-flow: column;
    -ms-flex-flow: column;
    flex-flow: column;
    -webkit-align-items: unset;
    -webkit-box-align: unset;
    -ms-flex-align: unset;
    align-items: unset;
  }
}

<div
    class="c0"
    label="Availability reserved"
  >
    <div
      class="c1"
      data-test-id="availability-reserved-label"
    >
      <span
        class="c2"
      >
        Availability reserved 
      </span>
    </div>
    <div
      class="c3"
      data-test-id="availability-reserved-value"
    >
      <span>
        $ 6,000,000
      </span>
    </div>
  </div>
  .c2 {
  display: inline-block;
  font-weight: bold;
}

.c0 {
  margin-top: 24px;
  margin-bottom: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0:last-child {
  margin-bottom: 0px;
}

.c1 {
  width: 250px;
  margin-right: 10px;
}

.c3 {
  display: inline-block;
}

@media (max-width:768px) {
  .c0 {
    -webkit-flex-flow: column;
    -ms-flex-flow: column;
    flex-flow: column;
    -webkit-align-items: unset;
    -webkit-box-align: unset;
    -ms-flex-align: unset;
    align-items: unset;
  }
}

<div
    class="c0"
    label="Minimum risk fee"
    value="0.90 %"
  >
    <div
      class="c1"
      data-test-id="minimum-risk-fee-label"
    >
      <span
        class="c2"
      >
        Minimum risk fee 
      </span>
    </div>
    <div
      class="c3"
      data-test-id="minimum-risk-fee-value"
    >
      0.90 %
    </div>
  </div>
  .c2 {
  display: inline-block;
  font-weight: bold;
}

.c0 {
  margin-top: 24px;
  margin-bottom: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0:last-child {
  margin-bottom: 0px;
}

.c1 {
  width: 250px;
  margin-right: 10px;
}

.c3 {
  display: inline-block;
}

@media (max-width:768px) {
  .c0 {
    -webkit-flex-flow: column;
    -ms-flex-flow: column;
    flex-flow: column;
    -webkit-align-items: unset;
    -webkit-box-align: unset;
    -ms-flex-align: unset;
    align-items: unset;
  }
}

<div
    class="c0"
    label="Maximum tenor"
    value="90 days"
  >
    <div
      class="c1"
      data-test-id="maximum-tenor-label"
    >
      <span
        class="c2"
      >
        Maximum tenor 
      </span>
    </div>
    <div
      class="c3"
      data-test-id="maximum-tenor-value"
    >
      90 days
    </div>
  </div>
  .c2 h3 {
  margin-bottom: 0px;
}

.c2 p {
  font-style: italic;
  width: 80%;
}

.c1.c1.c1.c1 h3 {
  font-weight: bold;
}

.c4 display:inline-block span {
  vertical-align: middle;
}

.c4 svg {
  margin-right: 10px;
  width: 14px;
  height: 14px;
  vertical-align: middle;
}

.c0 {
  border-top: 1px solid #c0cfde;
  padding: 20px 0;
}

.c0.c0 .column:not(.grid) {
  padding-bottom: 10px;
}

.c3 {
  margin-top: 16px;
}

<div
    class="c0"
  >
    <div
      class="c1 c2"
      data-test-id="credit-appetite-header-seller"
    >
      <h3>
        Ms. Seller information
      </h3>
      <p>
        Information you have shared with specific sellers
      </p>
    </div>
    <div
      class="c3"
    >
      <div
        class="ui stackable grid"
      >
        <div
          class="three column row"
        >
          <div
            class="column"
          >
            <div
              class="c4"
            >
              <svg
                data-test-id="shared-icon"
              >
                check-green.svg
              </svg>
              <span
                data-test-id="appetite-label"
              >
                Appetite
              </span>
            </div>
          </div>
          <div
            class="column"
          >
            <div
              class="c4"
            >
              <svg
                data-test-id="shared-icon"
              >
                check-green.svg
              </svg>
              <span
                data-test-id="availability-y-n-label"
              >
                Availability (Y/N)
              </span>
            </div>
          </div>
          <div
            class="column"
          >
            <div
              class="c4"
            >
              <svg
                data-test-id="shared-icon"
              >
                check-green.svg
              </svg>
              <span
                data-test-id="credit-limit-label"
              >
                Credit limit
              </span>
            </div>
          </div>
          <div
            class="column"
          >
            <div
              class="c4"
            >
              <svg
                data-test-id="shared-icon"
              >
                check-green.svg
              </svg>
              <span
                data-test-id="risk-fee-label"
              >
                Risk fee
                <span>
                   - 0.10 %
                </span>
              </span>
            </div>
          </div>
          <div
            class="column"
          >
            <div
              class="c4"
            >
              <svg
                data-test-id="shared-icon"
              >
                check-green.svg
              </svg>
              <span
                data-test-id="margin-label"
              >
                Margin
                <span>
                   - 100.00 %
                </span>
              </span>
            </div>
          </div>
          <div
            class="column"
          >
            <div
              class="c4"
            >
              <svg
                data-test-id="shared-icon"
              >
                check-green.svg
              </svg>
              <span
                data-test-id="max-tenor-label"
              >
                Max. tenor
                <span>
                   - 100 days
                </span>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CreditAppetiteInformationPanel should render correctly no credit information 1`] = `
<DocumentFragment>
  .c1 h3 {
  margin-bottom: 0px;
}

.c1 p {
  font-style: italic;
  width: 80%;
}

.c0.c0.c0.c0 h3 {
  font-weight: bold;
}

<div
    class="c0 c1"
    data-test-id="credit-appetite-header-empty"
  >
    <h3>
      No credit appetite information on Mr. Buyer
    </h3>
    <p>
      To enter information, go to the appetite section and add Mr. Buyer.
    </p>
  </div>
</DocumentFragment>
`;

exports[`CreditAppetiteInformationPanel should render correctly only Buyer information 1`] = `
<DocumentFragment>
  .c1 h3 {
  margin-bottom: 0px;
}

.c1 p {
  font-style: italic;
  width: 80%;
}

.c0.c0.c0.c0 h3 {
  font-weight: bold;
}

<div
    class="c0 c1"
    data-test-id="credit-appetite-header-buyer"
  >
    <h3>
      Mr. Buyer information
    </h3>
    <p>
      Internal information relating to the buyer
    </p>
  </div>
  .c2 {
  display: inline-block;
  font-weight: bold;
}

.c0 {
  margin-top: 24px;
  margin-bottom: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0:last-child {
  margin-bottom: 0px;
}

.c1 {
  width: 250px;
  margin-right: 10px;
}

.c3 {
  display: inline-block;
}

@media (max-width:768px) {
  .c0 {
    -webkit-flex-flow: column;
    -ms-flex-flow: column;
    flex-flow: column;
    -webkit-align-items: unset;
    -webkit-box-align: unset;
    -ms-flex-align: unset;
    align-items: unset;
  }
}

<div
    class="c0"
    label="Appetite"
    value="Yes"
  >
    <div
      class="c1"
      data-test-id="appetite-label"
    >
      <span
        class="c2"
      >
        Appetite 
      </span>
    </div>
    <div
      class="c3"
      data-test-id="appetite-value"
    >
      Yes
    </div>
  </div>
  .c2 {
  display: inline-block;
  font-weight: bold;
}

.c0 {
  margin-top: 24px;
  margin-bottom: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0:last-child {
  margin-bottom: 0px;
}

.c1 {
  width: 250px;
  margin-right: 10px;
}

.c3 {
  display: inline-block;
}

@media (max-width:768px) {
  .c0 {
    -webkit-flex-flow: column;
    -ms-flex-flow: column;
    flex-flow: column;
    -webkit-align-items: unset;
    -webkit-box-align: unset;
    -ms-flex-align: unset;
    align-items: unset;
  }
}

<div
    class="c0"
    label="Credit limit"
    value="$ 10,000,000"
  >
    <div
      class="c1"
      data-test-id="credit-limit-label"
    >
      <span
        class="c2"
      >
        Credit limit 
      </span>
    </div>
    <div
      class="c3"
      data-test-id="credit-limit-value"
    >
      $ 10,000,000
    </div>
  </div>
  .c2 {
  display: inline-block;
  font-weight: bold;
}

.c0 {
  margin-top: 24px;
  margin-bottom: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0:last-child {
  margin-bottom: 0px;
}

.c1 {
  width: 250px;
  margin-right: 10px;
}

.c3 {
  display: inline-block;
}

@media (max-width:768px) {
  .c0 {
    -webkit-flex-flow: column;
    -ms-flex-flow: column;
    flex-flow: column;
    -webkit-align-items: unset;
    -webkit-box-align: unset;
    -ms-flex-align: unset;
    align-items: unset;
  }
}

<div
    class="c0"
    label="Expiry date"
  >
    <div
      class="c1"
      data-test-id="expiry-date-label"
    >
      <span
        class="c2"
      >
        Expiry date 
      </span>
    </div>
    <div
      class="c3"
      data-test-id="expiry-date-value"
    >
      -
    </div>
  </div>
  .c2 {
  display: inline-block;
  font-weight: bold;
}

.c0 {
  margin-top: 24px;
  margin-bottom: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0:last-child {
  margin-bottom: 0px;
}

.c1 {
  width: 250px;
  margin-right: 10px;
}

.c3 {
  display: inline-block;
}

@media (max-width:768px) {
  .c0 {
    -webkit-flex-flow: column;
    -ms-flex-flow: column;
    flex-flow: column;
    -webkit-align-items: unset;
    -webkit-box-align: unset;
    -ms-flex-align: unset;
    align-items: unset;
  }
}

<div
    class="c0"
    label="Availability"
    value="Yes"
  >
    <div
      class="c1"
      data-test-id="availability-label"
    >
      <span
        class="c2"
      >
        Availability 
      </span>
    </div>
    <div
      class="c3"
      data-test-id="availability-value"
    >
      Yes
    </div>
  </div>
  .c2 {
  display: inline-block;
  font-weight: bold;
}

.c0 {
  margin-top: 24px;
  margin-bottom: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0:last-child {
  margin-bottom: 0px;
}

.c1 {
  width: 250px;
  margin-right: 10px;
}

.c3 {
  display: inline-block;
}

@media (max-width:768px) {
  .c0 {
    -webkit-flex-flow: column;
    -ms-flex-flow: column;
    flex-flow: column;
    -webkit-align-items: unset;
    -webkit-box-align: unset;
    -ms-flex-align: unset;
    align-items: unset;
  }
}

<div
    class="c0"
    label="Amount available"
  >
    <div
      class="c1"
      data-test-id="amount-available-label"
    >
      <span
        class="c2"
      >
        Amount available 
      </span>
    </div>
    <div
      class="c3"
      data-test-id="amount-available-value"
    >
      <span>
        $ 8,000,000
      </span>
    </div>
  </div>
  .c2 {
  display: inline-block;
  font-weight: bold;
}

.c0 {
  margin-top: 24px;
  margin-bottom: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0:last-child {
  margin-bottom: 0px;
}

.c1 {
  width: 250px;
  margin-right: 10px;
}

.c3 {
  display: inline-block;
}

@media (max-width:768px) {
  .c0 {
    -webkit-flex-flow: column;
    -ms-flex-flow: column;
    flex-flow: column;
    -webkit-align-items: unset;
    -webkit-box-align: unset;
    -ms-flex-align: unset;
    align-items: unset;
  }
}

<div
    class="c0"
    label="Availability reserved"
  >
    <div
      class="c1"
      data-test-id="availability-reserved-label"
    >
      <span
        class="c2"
      >
        Availability reserved 
      </span>
    </div>
    <div
      class="c3"
      data-test-id="availability-reserved-value"
    >
      <span>
        $ 6,000,000
      </span>
    </div>
  </div>
  .c2 {
  display: inline-block;
  font-weight: bold;
}

.c0 {
  margin-top: 24px;
  margin-bottom: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0:last-child {
  margin-bottom: 0px;
}

.c1 {
  width: 250px;
  margin-right: 10px;
}

.c3 {
  display: inline-block;
}

@media (max-width:768px) {
  .c0 {
    -webkit-flex-flow: column;
    -ms-flex-flow: column;
    flex-flow: column;
    -webkit-align-items: unset;
    -webkit-box-align: unset;
    -ms-flex-align: unset;
    align-items: unset;
  }
}

<div
    class="c0"
    label="Minimum risk fee"
    value="0.90 %"
  >
    <div
      class="c1"
      data-test-id="minimum-risk-fee-label"
    >
      <span
        class="c2"
      >
        Minimum risk fee 
      </span>
    </div>
    <div
      class="c3"
      data-test-id="minimum-risk-fee-value"
    >
      0.90 %
    </div>
  </div>
  .c2 {
  display: inline-block;
  font-weight: bold;
}

.c0 {
  margin-top: 24px;
  margin-bottom: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0:last-child {
  margin-bottom: 0px;
}

.c1 {
  width: 250px;
  margin-right: 10px;
}

.c3 {
  display: inline-block;
}

@media (max-width:768px) {
  .c0 {
    -webkit-flex-flow: column;
    -ms-flex-flow: column;
    flex-flow: column;
    -webkit-align-items: unset;
    -webkit-box-align: unset;
    -ms-flex-align: unset;
    align-items: unset;
  }
}

<div
    class="c0"
    label="Maximum tenor"
    value="90 days"
  >
    <div
      class="c1"
      data-test-id="maximum-tenor-label"
    >
      <span
        class="c2"
      >
        Maximum tenor 
      </span>
    </div>
    <div
      class="c3"
      data-test-id="maximum-tenor-value"
    >
      90 days
    </div>
  </div>
</DocumentFragment>
`;
